pkg/storage/etcd/etcdcommon/config.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdcommon
```

Comment:
Should setup for non-etcd backends (e.g. people are threatening to add consul) go in this file/package or elsewhere? I think probably this file? and therefore I think this package should be e.g. pkg/storage/factory or some such.